polaris/hub/client.py

```diff
@@ -359,6 +359,9 @@ def open_zarr_file(
 
         try:
             store = zarr.storage.FSStore(path, fs=polaris_fs)
+            if mode in ["r", "r+"]:
+                zarr.consolidate_metadata(store)
+                return zarr.open_consolidated(store, mode=mode)
```

Comment:
No need to call `open_consolidated` again, since the return value of `zarr.consolidate_metadata` is the root opened in consolidated mode. See [the source code](https://zarr.readthedocs.io/en/stable/_modules/zarr/convenience.html#consolidate_metadata).